src/main/java/app/linguistai/bmvp/service/wordbank/WordSelectionService.java

```diff
@@ -0,0 +1,147 @@
+package app.linguistai.bmvp.service.wordbank;
+
+import java.time.LocalDate;
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.List;
+import java.util.UUID;
+
+import app.linguistai.bmvp.exception.NotFoundException;
+import app.linguistai.bmvp.exception.SomethingWentWrongException;
+import app.linguistai.bmvp.repository.wordbank.IUnknownWordRepository;
+import app.linguistai.bmvp.repository.wordbank.IWordSelectionRepository;
+import org.springframework.stereotype.Service;
+
+import app.linguistai.bmvp.model.User;
+import app.linguistai.bmvp.model.enums.ConfidenceEnum;
+import app.linguistai.bmvp.model.wordbank.UnknownWord;
+import app.linguistai.bmvp.model.wordbank.WordSelection;
+import app.linguistai.bmvp.repository.IAccountRepository;
+import app.linguistai.bmvp.request.QSelectWord;
+import lombok.RequiredArgsConstructor;
+import lombok.extern.slf4j.Slf4j;
+
+@Slf4j
+@RequiredArgsConstructor
+@Service
+public class WordSelectionService {
+
+    private final IWordSelectionRepository wordSelectionRepository;
+    private final IAccountRepository accountRepository;
+    private final IUnknownWordRepository unknownWordRepository;
+
+    public List<WordSelection> getSelectedWords(QSelectWord selectWord, String email) throws Exception {
+        try {
+            User user = accountRepository.findUserByEmail(email)
+                .orElseThrow(() -> new NotFoundException("User does not exist for given email: [" + email + "]."));
+
+
+            List<String> preservedWords = new ArrayList<>(selectWord.getPreservedWords());
+
+            // Get the previously selected words
+            List<WordSelection> selectedWords = 
+                                wordSelectionRepository
+                                .findByConversationIdAndWordOwnerListUserEmail(
+                                    selectWord.getConversationId(), 
+                                    user.getEmail());
+
+            // If words are not selected yet, select them and save to db
+            if (selectedWords.isEmpty()) {
+                List<UnknownWord> selectedUnknownWords = selectWords(user.getId(), selectWord.getSize());
+
+                // Iterate over each selected word and create a WordSelection object
+                for (UnknownWord unknownWord : selectedUnknownWords) {
+                    WordSelection wordSelection = WordSelection.builder()
+                            .conversationId(selectWord.getConversationId())
+                            .word(unknownWord)
+                            .build();
+
+                    selectedWords.add(wordSelection);
+                }
+
+                // Save the selected words to db
+                wordSelectionRepository.saveAll(selectedWords);
+
+                log.info("New words are selected for user email {}", email);
+            } else {
```

Comment:
This else is not needed if we directly return after the if block before. This will reduce nesting and increase readability and maintainability of the code

Reply:
Done